spec/helpers/params_spec.rb

```diff
@@ -34,4 +34,18 @@ def app
     post "/", "<hello>world</hello>", {"CONTENT_TYPE" => "application/xml"}
     assert_equal "{}", last_response.body
   end
+
+  it "should throw bad request when receiving invalid json via post" do
+    assert_raises(Pliny::Errors::BadRequest) do
+      post "/", "{\"foo\"}", {"CONTENT_TYPE" => "application/json"}
+    end
+  end
+
+  it "should throw bad request and the correct message when post has invalid json" do
+    begin
+      post "/", "{\"foo\"}", {"CONTENT_TYPE" => "application/json"}
+    rescue Pliny::Errors::BadRequest => e
+      assert_match /unexpected token/, e.message
+    end
+  end
```

Comment:
Looking at these a second time, I'm wondering if we could/should combine these into a single case. Thoughts?

```suggestion
  it "should throw bad request when receiving invalid json via post" do
    err = assert_raises(Pliny::Errors::BadRequest) do
      post "/", "{\"foo\"}", {"CONTENT_TYPE" => "application/json"}
    end

    assert_match /unexpected token/, err.message
  end
```

Reply:
Good thinking +1